src/cargo/util/command_prelude.rs

```diff
@@ -1404,6 +1405,30 @@ fn get_packages() -> CargoResult<Vec<Package>> {
     Ok(packages)
 }
 
+pub fn get_direct_dependencies_pkg_name_candidates() -> Vec<clap_complete::CompletionCandidate> {
+    let dependencies = get_dependencies_from_metadata();
+    dependencies
+        .unwrap_or_default()
+        .into_iter()
+        .map(|dependency| dependency.package_name().to_string().into())
+        .collect::<Vec<_>>()
+}
+
+fn get_dependencies_from_metadata() -> CargoResult<Vec<Dependency>> {
+    let cwd = std::env::current_dir()?;
+    let gctx = GlobalContext::new(shell::Shell::new(), cwd.clone(), cargo_home_with_cwd(&cwd)?);
+    let ws = Workspace::new(&find_root_manifest_for_wd(&cwd)?, &gctx)?;
+
+    let dependencies = ws
+        .members()
+        .flat_map(|pkg| pkg.dependencies().into_iter().cloned())
+        .collect::<HashSet<_>>()
+        .into_iter()
+        .collect::<Vec<_>>();
```

Comment:
Since we put this through a `HashSet`, should we do a `dependencies.sort()`?

Reply:
We could also do a fancier sort, like put deps within the current package first and then put deps for all workspace members.  Thoughts?